addons/amazon-s3-online-storage/index.js

```diff
@@ -0,0 +1,178 @@
+            }
+            resolve();
+          });
+        } else {
+          reject(error)
+        }
+      });
+    });
+  }
+
+  _ensureBatch() {
+    if (!this.batchAppend || !this.uploader) {
+      this.files = [];
+      return this._newBatch();
+    }
+    return Promise.resolve();
+  }
+
+  _initCredentials(options) {
+    AWS.config.update({
+      credentials: new AWS.Credentials(options.awsSecretKeyId, options.awsSecretAccessKey, options.awsSessionToken),
+      region: options.region,
+      useAccelerateEndpoint: options.useS3Accelerate || false
+    });
+
+    AWS.config.credentials.expireTime = new Date(options.expiration);
+  }
+
+  _newBatch() {
+    var resource = this._resource();
+    resource.path = 'upload/new/s3';
+    return resource.post().then((response) => {
+      this.batchId = response.batchId;
+      this.extraInfo = response.extraInfo;
+      this._initCredentials(response.extraInfo);
+      this.uploader = new AWS.S3({
+        params: {
+          Bucket: this.extraInfo.bucket,
+        },
+        computeChecksums: true
+      });
+    });
+  }
+
+  get accepts() {
+    return UploaderBehavior.getProviders()['default'].prototype.accepts;
+  }
+
+  upload(files, callback) {
+    this._ensureBatch().then(() => {
+      if ((new Date()).getTime() >= this.extraInfo.expiration) {
+        this._refreshBatchInfo();
+      }
+
+      var promises = [];
+      for (var i = 0; i < files.length; ++i) {
+        var file = files[i];
+        file.index = this.files.length;
+        this.files.push(file);
+        promises.push(this._upload(file, callback));
+      }
+      return Promise.all(promises).then(() => {
+        callback({ type: 'batchFinished', batchId: this.batchId });
+      });
+    }).catch((error) => {
+      callback({ type: 'uploadInterrupted', error });
+    });
+  }
+
+  hasProgress() {
+    return true;
+  }
+
+  cancelBatch() {
+    if (this.uploader) {
+      for (var i = 0; i < this.files.length; ++i) {
+        var file = this.files[i];
+        if (file.managedUpload) {
+          file.managedUpload.abort();
+        }
+      }
+    }
+    this.uploader = null;
+    this.batchId = null;
+    this.files = [];
+  }
+
+  batchExecute(operationId, params, headers) {
+    return this.connection.operation(operationId).then((operation) => {
+      var options = {
+        url: [operation._nuxeo._restURL, 'upload', this.batchId, 'execute', operationId].join('/')
+            .replace(/\/+/g, '\/')
+      };
+      if (headers) {
+        options['headers'] = headers;
+      }
+      if (params.context) {
+        operation = operation.context(params.context);
+      }
+      return operation.params(params).execute(options);
+    });
+  }
+}
+
+UploaderBehavior.registerProvider('s3', S3Provider);
+
+// if S3 direct upload is enabled set it as default upload provider
+// config values from configuration service are strings
+if (String(Nuxeo.UI.config.s3.useDirectUpload) == 'true') {
```

Comment:
use strict equals instead:
```suggestion
if (String(Nuxeo.UI.config.s3.useDirectUpload) === 'true') {
```

Reply:
Since we're effectively "casting" to `String` there's no point in doing the strict equals, ie `String(1) === '1'` 🤷‍♂️